File: orchestration/templates/tests/typescript/example.test.ts
/**
 * Example TypeScript Test Template
 *
 * Replace this with actual tests for your component.
 */

describe('Example Test Suite', () => {
  test('example test - replace with actual tests', () => {
    expect(true).toBe(true);
  });

  test('string operations', () => {
    const result = 'hello' + ' ' + 'world';
    expect(result).toBe('hello world');
    expect(result.length).toBe(11);
  });

  test('array operations', () => {
    const items = [1, 2, 3];
    items.push(4);
    expect(items).toHaveLength(4);
    expect(items[items.length - 1]).toBe(4);
  });

  test('object operations', () => {
    const data = { key: 'value' };
    data['newKey'] = 'newValue';
    expect(data).toHaveProperty('key');
    expect(data.newKey).toBe('newValue');
  });
});

// TODO: Replace these examples with actual tests
// Example test structure:
//
// describe('ComponentName', () => {
//   test('should do something', () => {
//     // Arrange
//     const input = setupTestData();
//
//     // Act
//     const result = functionUnderTest(input);
//
//     // Assert
//     expect(result).toBe(expectedValue);
//   });
// });

File: orchestration/templates/tests/go/example_test.go
// Example Go Test Template
//
// Replace this with actual tests for your component.

package main

import (
	"testing"
)

func TestExample(t *testing.T) {
	// Example test - replace with actual tests
	if false {
		t.Error("Example test failed")
	}
}

func TestStringOperations(t *testing.T) {
	result := "hello" + " " + "world"
	expected := "hello world"

	if result != expected {
		t.Errorf("Expected '%s', got '%s'", expected, result)
	}

	if len(result) != 11 {
		t.Errorf("Expected length 11, got %d", len(result))
	}
}

func TestSliceOperations(t *testing.T) {
	items := []int{1, 2, 3}
	items = append(items, 4)

	if len(items) != 4 {
		t.Errorf("Expected length 4, got %d", len(items))
	}

	if items[len(items)-1] != 4 {
		t.Errorf("Expected last element to be 4, got %d", items[len(items)-1])
	}
}

// TODO: Replace these examples with actual tests
// Example test structure:
//
// func TestFunctionName(t *testing.T) {
//     // Arrange
//     input := setupTestData()
//
//     // Act
//     result := functionUnderTest(input)
//
//     // Assert
//     if result != expectedValue {
//         t.Errorf("Expected %v, got %v", expectedValue, result)
//     }
// }
